code/modules/food_and_drinks/food.dm

```diff
@@ -56,3 +72,75 @@
 	if(ant_timer)
 		deltimer(ant_timer)
 	ant_timer = addtimer(CALLBACK(src, .proc/check_for_ants), 3000, TIMER_STOPPABLE)
+
+/obj/item/reagent_containers/food/proc/checkLiked(var/fraction, mob/M)
+	if(last_check_time + 50 < world.time)
+		if(ishuman(M))
+			var/mob/living/carbon/human/H = M
+			if(foodtype & H.dna.species.toxic_food)
+				var/type_string = matched_type(foodtype & H.dna.species.toxic_food)
+				to_chat(H,"<span class='warning'>[toxic_message(type_string)]</span>")
+
+				H.AdjustDisgust(25 + 30 * fraction)
+			else if(foodtype & H.dna.species.disliked_food)
+				var/type_string = matched_type(foodtype & H.dna.species.disliked_food)
+				to_chat(H,"<span class='notice'>[dislike_message(type_string)]</span>")
+
+				H.AdjustDisgust(11 + 15 * fraction)
+			else if(foodtype & H.dna.species.liked_food)
+				var/type_string = matched_type(foodtype & H.dna.species.liked_food)
+				to_chat(H,"<span class='notice'>[love_message(type_string)]</span>")
+
+				H.AdjustDisgust(-5 + -2.5 * fraction)
+			last_check_time = world.time
+
+/obj/item/reagent_containers/food/proc/toxic_message(type)
+	var/plural = cmptext(type[length(type)], "s") ? "are" : "is"
+
+	return pick(
+		"What the hell was that?! I hate <b>[type]</b>!",
+		"That was awful! I hate <b>[type]</b>.",
+		"God, that was outright dangerous! <b>[capitalize(type)]</b> [plural] awful!")
+
+/obj/item/reagent_containers/food/proc/dislike_message(type)
+	var/plural = cmptext(type[length(type)], "s") ? "are" : "is"
+
+	return pick(
+		"That wasn't very good. I should probably stay away from <b>[type]</b>.",
+		"Not a fan of <b>[type]</b>. Let's not eat that again.",
+		"Eugh. <b>[capitalize(type)]</b> really [plural]n't something I should be eating.")
+
+/obj/item/reagent_containers/food/proc/love_message(type)
```

Comment:
Can combine all of these into one proc and instead of just giving the type you can give the list of possible answers.
Be sure to use defines for that though